ignite/handlers/ema_handler.py

```
@@ -0,0 +1,207 @@
+        denominator = max(1, self.warmup_iters - 1)
+        momentum = self.momentum_start + (self.momentum_end - self.momentum_start) * (curr_iter - 1) / denominator
+        return min(self.momentum_end, momentum)
+
+    def _update_ema_model(self, engine: Engine, name: str) -> None:
+        """Update weights of ema model"""
+        momentum = getattr(engine.state, name)
+        for ema_v, model_v in zip(self.ema_model.state_dict().values(), self.model.state_dict().values()):
+            print(name, ema_v.data, model_v.data)
+            ema_v.mul_(1.0 - momentum).add_(model_v.data, alpha=momentum)
+
+    def _update_ema_momentum(self, engine: Engine, name: str) -> None:
+        """Update momentum in engine.state"""
+        curr_iter = engine.state.iteration
+        momentum = self._get_momentum(curr_iter)
+        setattr(engine.state, name, momentum)
+
+    def attach(
+        self,
+        engine: Engine,
+        name: str = "ema_momentum",
+        event: Union[str, Events, CallableEventWithFilter, EventsList] = Events.ITERATION_COMPLETED,
+    ) -> None:
+        """Attach the handler to engine. After the handler is attached, the ``Engine.state`` (a
+        ``dict``) will add an new attribute with name ``name``. Then, current momentum can be retrieved by
+        from ``Engine.state`` when the engine runs.
+
+        Args:
+            engine: trainer to which the handler will be attached.
+            name: attribute name for retrieving EMA momentum from ``Engine.state``. It should be a unique name since a
+                trainer can have multiple EMA handlers.
+            event: event when the EMA momentum and EMA model are updated.
+
+        """
+        if hasattr(engine.state, name):
+            warnings.warn(
+                f"Attribute: '{name}' is already in Engine.state. Thus it might be "
+                f"overridden by other EMA handlers. Please select another name.",
+                UserWarning,
+            )
```

In your previous commits with dict, I think you raised an error instead of the warning. I think it is good to stop execution instead of sliently overwritting things ...